presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -102,10 +104,9 @@ public FunctionListBuilder window(String name, Class<? extends ValueWindowFuncti
                 WINDOW,
                 ImmutableList.of(typeVariable(typeVariable)),
                 ImmutableList.of(),
-                typeVariable,
-                ImmutableList.copyOf(argumentTypes),
-                false,
-                ImmutableSet.of());
+                parseTypeSignature(typeVariable),
+                Arrays.asList(argumentTypes).stream().map(x -> parseTypeSignature(x)).collect(toImmutableList()),
```

Comment:
You should be able to use a method reference in lieu of the lambda. Something like `.map(TypeSignature::parseTypeSignature)`